apps/extension/src/state/tx-approval.ts

```
@@ -117,31 +124,27 @@ export const createTxApprovalSlice =
     },
 
     sendResponse: () => {
-      const {
-        responder,
-        choice,
-        transactionView: transactionViewString,
-        authorizeRequest: authorizeRequestString,
-      } = get().txApproval;
+      const { responder, choice, authorizeRequest, invalidPlan } = get().txApproval;
 
       try {
         if (!responder) {
-          throw new Error('No responder');
+          throw new ReferenceError('No responder');
         }
 
         try {
-          if (choice === undefined || !transactionViewString || !authorizeRequestString) {
-            throw new Error('Missing response data');
+          if (invalidPlan) {
+            throw invalidPlan;
```

if a validation failure is present upon `sendResponse`, it's always thrown instead of returning a choice